src/components/Content/Discounts/Discounts.tsx

```diff
@@ -3,15 +3,50 @@ import './discounts.scss';
 import Card from '../../Card/Card';
 import { usePageChanger } from '../../../utils';
 import { Link } from 'react-router-dom';
+import { Phone } from '../../../types/Phone';
 
-export default function Discounts() {
+interface DiscountsProps {
+  phones: Phone[];
+}
+
+export default function Discounts({ phones }: DiscountsProps) {
 	const page = usePageChanger(1);
-	const { currentCardPag, showingCards, firstPage, lastPage, } = page;
+	const { currentCardPag, firstPage } = page;
+
+	const getModelName = (name: string) => {
+		const regex = /^(.+)\s\d+GB/;
+		const match = name.match(regex);
+		return match ? match[1] : name;
+	};
+
+	const uniquePhones = Array.from(
+		phones
+			.reduce((map, phone) => {
+				const modelName = getModelName(phone.name);
+				if (!map.has(modelName)) {
+					map.set(modelName, phone);
+				}
+				return map;
+			}, new Map())
+			.values()
+	);
+
+	const discountedPhones = uniquePhones.filter(
+		(phone) => phone.price < phone.fullPrice
+	);
+
+	const shuffledPhones = [...discountedPhones].sort(() => Math.random() - 0.5);
+
+	const CARDS_PER_PAGE = 4;
+	const totalPages = Math.ceil(shuffledPhones.length / CARDS_PER_PAGE);
+	const startIndex = (currentCardPag - 1) * CARDS_PER_PAGE;
+	const endIndex = startIndex + CARDS_PER_PAGE;
+	const paginatedPhones = shuffledPhones.slice(startIndex, endIndex);
```

Comment:
w tym miejscu i w NewModels powtarza się kod, można to wynieść w jakąś funkcję będzie lakonicznie. Możesz pokombinować z kodem z utilsa - może jakoś go updatować, żeby się nie powtarzać.